### DISCUSSION TITLE
read end reason

### DISCUSSION THREAD

**Discussion by ylang0825** [LooseLab/readfish#238] [Q&A] [answered]
Hi everyone!
I have successfully used readfish to enrich the target and I'm quite confused about the reads splitted by end reasons. After running readfish, In the MinKNOW UI, I observed a purple bar emerged with the "Adaptive sampling voltage reversal" label, which I think reflects readfish's unblocking. However, there is also a high red bar labeled "Read became blocked" and a tiny orange bar labeled "Unblock voltage reversal". I wonder what these two labels means? Are they also produced by readfish? Since it seems these two bars did not appear in non-adaptive sequencing.

By the way, is there any difference between readfish and on-board MinKNOW adaptive sequencing? It seems that they use the same strategy. Or they may use different parameters?

Thanks very much!

**Comment by mattloose** [comment, marked as answer]
You are correct that the purple bar labelled "Adaptive sampling voltage reversal" is the action of ReadFish's unblocking. 

The label "read became blocked" and "unblock voltage reversal" are classifications that may be caused by either a bit of DNA blocked in the pore or by the activity of ReadFish itself. Because the classification of reads is so fast it is possible that errors can be made.

Those states do appear during normal sequencing - the only one that is specific to adaptive sampling is "Adaptive sampling voltage reversal". On a library that is not particuarly blocky you won't see many of the other categories but they are there.

The difference between readfish and onboard MinKNOW adaptive sequencing is the granularity of control that you can get with readfish. Essentially readfish is more configurable than standard adaptive sampling and you can do things like change the reference and targets during sequencing (which currently you can't do with on-board MinKNOW adaptive sequencing). If you are running a standard fixed panel then you should work with the built in adaptive sampling. If you want to do more sophisticated things like barcodes with different panels or other iterative adaptive sampling experiments something like readfish is more suitable.

hope that helps.
